zebra-consensus/src/verify/block.rs

```diff
@@ -471,4 +492,71 @@ mod tests {
 
         Ok(())
     }
+
+    #[tokio::test]
+    #[spandoc::spandoc]
+    async fn coinbase() -> Result<(), Report> {
+        install_tracing();
+
+        // Service variables
+        let state_service = Box::new(zebra_state::in_memory::init());
+        let mut block_verifier = super::init(state_service);
+
+        // Get a header of a block
+        let header =
+            BlockHeader::zcash_deserialize(&zebra_test::vectors::DUMMY_HEADER[..]).unwrap();
+
+        let ready_verifier_service = block_verifier.ready_and().await.map_err(|e| eyre!(e))?;
+
+        /// Test 1: Empty transaction
+        let block = Block {
+            header,
+            transactions: Vec::new(),
+        };
+
+        /// Error: no coinbase transaction in block
+        ready_verifier_service
+            .call(Arc::new(block.clone()))
+            .await
+            .unwrap_err();
+
+        let ready_verifier_service = block_verifier.ready_and().await.map_err(|e| eyre!(e))?;
+
+        /// Test 2: Transaction at first position is not coinbase
```

Comment:
you seem to be mixing doc comments and regular comments, I'm not sure if you're familiar with spandoc but I'm going to assume you're not so let me explain why we use doc comments in our tests. `spandoc` is a proc macro library that converts doc comments within a function to [tracing spans](https://docs.rs/tracing/0.1.15/tracing/span/index.html) around the documented expression. This means the following two code snippets are equivalent

```rust
        /// Test 2: Transaction at first position is not coinbase
        let mut transactions = Vec::new();
```

```rust
        let span = span!(tracing::Level::ERROR, "coinbase::comment", text="Test 2: Transaction at first position is not coinbase");
        let guard = span.enter();
        let mut transactions = Vec::new();
        drop(guard);
        drop(span);
```

So this instrumentation that you've created with the doc comment is wrapping an expression that as far as I know cannot panic, throw an error, or emit any tracing events, meaning the span is never used.

If you want to instrument all of the expressions here you should wrap them in a block

```rust
        /// Test 2: Transaction at first position is not coinbase
        {
        let mut transactions = Vec::new();
        let tx = Transaction::zcash_deserialize(&zebra_test::vectors::DUMMY_TX1[..]).unwrap();
        transactions.push(Arc::new(tx));
        let block = Block {
            header,
            transactions,
        };
        }
```

Reply:
All these function calls can't fail.

I think we need to end the span block after awaiting the future:
```
ready_verifier_service
            .call(Arc::new(block))
            .await
            .unwrap_err();
```

Or just use a doc comment above the future.

Reply:
spandoc changes moved to https://github.com/ZcashFoundation/zebra/issues/609